### ISSUE TITLE
Support for mac catalyst with apple silicon.

### ISSUE THREAD

**Issue by kasimok** [Azure-Samples/cognitive-services-speech-sdk#1948]
**Describe the bug**
I see catalyst is somehow supported, but not fully supported.
Run the sample on Mac Catalyst on x86 macOS works but on apple silicon it shows error of `No such module 'MicrosoftCognitiveServicesSpeech'`

**To Reproduce**
1. Add `Mac Catalyst` to the supported destinations for the `speech-sample` project
2. Runs OK on x86_64 Macs but framework not found on arm Macs.

**Expected behavior**
fully support Mac Catalyst for both x86_64 and arm mac

**Version of the Cognitive Services Speech SDK**
1.28

**Platform, Operating System, and Programming Language**
 - OS: macOS 13.2
 - Hardware - Apple M1 Max
 - Programming language: Swift
 - Browser NA

**Additional context**
Why don't we support SPM directly?

**Comment by jhakulin** [comment]
@kasimok Thanks for the report. https://aka.ms/csspeech/iosbinary xcframework package do contains support for both x86_64 and arm64 (ios-arm64_x86_64-maccatalyst). 

Anyways, I will try to reproduce your problem.

**Comment by kasimok** [comment]
> @kasimok Thanks for the report. https://aka.ms/csspeech/iosbinary xcframework package do contains support for both x86_64 and arm64 (ios-arm64_x86_64-maccatalyst).
> 
> Anyways, I will try to reproduce your problem.

Yes, I noticed the file's name contains both arm64 and x86_64, but somehow, it won't let me use that framework on arm64 Mac Catalyst.

**Comment by jhakulin** [comment]
So far just verified that package contains required bits, not clear why you could not use it, so will be checking a bit more.

To see what is bundled inside the binary you can check:

% file MicrosoftCognitiveServicesSpeech

MicrosoftCognitiveServicesSpeech: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit dynamically linked shared library x86_64] [arm64]

MicrosoftCognitiveServicesSpeech (for architecture x86_64): Mach-O 64-bit dynamically linked shared library x86_64

MicrosoftCognitiveServicesSpeech (for architecture arm64): Mach-O 64-bit dynamically linked shared library arm64

**Comment by kasimok** [comment]
@jhakulin could you please verify with [your sample project](https://github.com/Azure-Samples/cognitive-services-speech-sdk/tree/master/samples/swift/ios/speech-samples), try adding catalysts support, can you run the Mac catalyst?

I create this issue as my colleague who have a x86_64 runs OK.


<img width="1982" alt="image" src="https://github.com/Azure-Samples/cognitive-services-speech-sdk/assets/5194603/3bf8f76c-d128-4eaa-af2e-b822434bc97a">

**Comment by kasimok** [comment]
@jhakulin Hi Jarno, did you tried it?

**Comment by jhakulin** [comment]
@kasimok I can repro the issue with the Swift sample you pointed out, thanks for the report. The issue might be related to pod installation. Because when I checked this sample https://github.com/Azure-Samples/cognitive-services-speech-sdk/tree/master/samples/objective-c/ios/speech-samples, it builds and works ok for Mac Catalyst target on my Mac M1 (ARM based architecture).

**Comment by kasimok** [comment]
@jhakulin I tried xcframework in the first place, the same behaviour. Our team does not use `cocoapods`, then we repacked it into `spm`(Swift Package Manager) since you provided the xcframework version. I order to use Azure Speech Service we have to install the cocoapods back to verify if it is a xcframework only issue. So in a word, `cocoapods`/`xcframework`/`spm`, they all behaves the same. 



Below is the package file I use in swift package manager
```
// swift-tools-version:5.5
// The swift-tools-version declares the minimum version of Swift required to build this package.

import PackageDescription

// swiftlint: disable prefixed_toplevel_constant
let package = Package(
  name: "MicrosoftCognitiveServicesSpeech",
  products: [
    // Products define the executables and libraries a package produces, and make them visible to other packages.
    .library(
      name: "MicrosoftCognitiveServicesSpeech",
      targets: ["MicrosoftCognitiveServicesSpeechTarget"])
  ],
  dependencies: [
    // Dependencies declare other packages that this package depends on.
    // .package(url: /* package url */, from: "1.0.0"),
  ],
  targets: [
    // Targets are the basic building blocks of a package. A target can define a module or a test suite.
    // Targets can depend on other targets in this package, and on products in packages this package depends on.
    .target(
      name: "MicrosoftCognitiveServicesSpeechTarget",
      dependencies: [
        .target(name: "MicrosoftCognitiveServicesSpeech")
      ],
      path: "MicrosoftCognitiveServicesSpeechTarget"),
    .binaryTarget(
      name: "MicrosoftCognitiveServicesSpeech",
      url: "https://csspeechstorage.blob.core.windows.net/drop/1.28.0/MicrosoftCognitiveServicesSpeech-XCFramework-1.28.0.zip",
      checksum: "4171300d3c0d198eb022ee95835560e800df47be591f5372c9a6ef0ebe53d2c5")
  ]
)
// swiftlint: enable prefixed_toplevel_constant
```

**Comment by kasimok** [comment]
> @kasimok Yeah, not clear what is the reason yet. Note that Objective-C project works with the Macatalyst and same xcframework. I will investigate why Swift project does not.

this maybe the clue: I compared other's fat xcframeworks that supports catalyst, they have 
```xml
<key>LSMinimumSystemVersion</key> 
<string>10.13</string>
 <key>CFBundleSupportedPlatforms</key>
 <array> <string>MacOSX</string> </array>
```
in the info.plist, they are missing in your published xcframework.
Also, the `MinimumOSVersion` is set to 13.1 on catalyst, which does not match the iOS's API level.

**Comment by jhakulin** [comment]
@kasimok Thank you for sharing the workaround, indeed using bridging header provides alternative way to module import. I will look into module import issue.

Internal workitem ref: 5276384

**Comment by jhakulin** [comment]
Potentially related issue: https://github.com/leetal/ios-cmake/issues/172

We will continue investigating this, however at the moment there seems to be an external issue between CMAKE and Xcode generation with iOS toolchain to support properly Maccatalyst builds with Xcode. After those are solved, I am hopeful there would be better resolution to Maccatalyst module import issue.

Using bridging header is recommended workaround at the moment.

**Comment by ericyangpoxst** [comment]
Ouch! Spent whole day on debugging this issue on my Apple Silicon! Can you guys update the README or some where at the document to highlight this issue till it's solved? It will save us a lot time! thx!

**Comment by kasimok** [comment]
@jhakulin I knew it now, your "CMAKE and Xcode generation with iOS toolchain" must missed the modulemap file for mac catalyst platform. That's why we cannot use the symbols in swift directly.

I tried, copy the modulemap, and the catalyst platform is OK.

<img width="1032" alt="Screenshot 2023-09-24 at 3 51 23 PM" src="https://github.com/Azure-Samples/cognitive-services-speech-sdk/assets/5194603/8e6cfdba-eaae-4c9f-a1de-77bec52bcc5e">

**Comment by jhakulin** [comment]
@kasimok Thank you for the information, I'll create a fix to next Speech SDK release. Thanks again!

**Comment by jhakulin** [comment]
The fix will be in the 1.33.0 Speech SDK release, closing the issue.
